contracts/smart-account/modules/MultichainValidator.sol

```diff
@@ -47,27 +39,36 @@ contract MultichainECDSAValidator is EcdsaOwnershipRegistryModule {
 
```

Comment:
Add the validUntil and validAfter field in non-multichain signatures as well.

Reply:
We currently do not use validUntil and validAfter for single chain (regular) ecdsa signatures.

I suggest not to do it. 

1) In most cases it is not needed, as the signature is over the userOp hash and the userOp is immediately sent to the bundler to be handled (like txns by EOA, which are signed and immediately sent to mempool), so in 99% of cases I suppose that validUntil and validAfter will be expected to be unlimited.

2) it is not expected that validUntil and validAfter are provided from outside now. As per 4337 single chain signatures are expected to be over userOpHash. If we need to provide validUntil and validAfter from outside we will have to append it to userOpHash and sign, so they can't be changed. Then we will need to unpack them on-chain and spend gas for it.

3) IMO, the idea for validUntil and validAfter is that it is not just passed to the SA from outside, but is defined by the SA (or in our case validation module) itself. Depending on the module specific.
For example , that's how it is done is Account Recovery Module
First we detect validity timeframe using some module specific logic => https://github.com/bcnmy/scw-contracts/blob/8fbe98723639dd0f21c6703de8149746f1d65ed2/contracts/smart-account/modules/SocialRecoveryModule.sol#L211-L219
Then we pack it => https://github.com/bcnmy/scw-contracts/blob/8fbe98723639dd0f21c6703de8149746f1d65ed2/contracts/smart-account/modules/SocialRecoveryModule.sol#L250-L253

Reply:
If user doesn't sign validUntil and validAfter, it opens up a situation where user signature is valid indefinitely, if some bundler decides to hold off an userOp (could be a swap userOp) and decides to execute it later (may be when it gives max returns to bundler somehow).. the user account could be affected in a wrong way.

Reply:
The same could have been valid for pre-AA txns/mempool. Also such a censorship may occur not only because of the economical interest of the bundler/validator, but for other reasons as well. 
However, since there are many actors, this system is censorship-resistant. 
The same should be valid for AA userOps pool.

According to this writeup by Yoav, 
https://notes.ethereum.org/@yoav/unified-erc-4337-mempool
One of the conditions of making erc4337 infra censorship-resistant
is that "Users can propagate user-operations to a network where any bundler can serve them."

What I'm trying to say is that if there is no censorship-resistant system, featuring the network of independent bundlers, just having validUntil/validAfter won't be enough to protect users from userOps censorship.

Thus, I believe that such a network of independent Bundlers will be in place, and if user chooses to use the private bundlers, those bundlers will be selected with a criteria of not profiting on their users (like flashbots do not benefit from txns processed).

Assuming this I still suggest not to introduce validUntil/validAfter for regular ecdsa signatures over userOpHash'es, which act more like a standard private key signatures over pre-AA txn.

Reply:
if we're passing validUntil, validAfter we can use _packValidationData() from Helpers class (infinitism) regardless. (sig failed / sig passed)

Reply:
Got it.. yeah in a censorship-resistant system, we wouldn't need to worry about it.